test/utils/wasm_engine.hpp

```diff
@@ -55,6 +56,17 @@ class WasmEngine
     /// Executes the function of the given index.
     /// Requires instantiate().
     virtual Result execute(FuncRef func_ref, const std::vector<uint64_t>& args) = 0;
+
+    static void validate_function_signature(std::string_view signature)
```

Comment:
I would prefer to make it a free function in `fizzy::test`, but not so important

Reply:
Would that mean to make inline in the header or creating yet another file, `wasm_engine.cpp`, to place it?

Reply:
@chfast any preference?

a) as it is now
b) as a free function in the header
c) as a free function in a new `.cpp` file

Reply:
For sanity, it is preferable to move implementation to .cpp file.
For static method / free function I think both options are fine.